app/server.py

```diff
@@ -43,30 +45,48 @@ def bad_request(error):
 
 @app.route('/api/v1/search/<search_engine>', methods=['GET'])
 def search(search_engine):
+    print(search_engine)
+    # convert search_engine into engine which should be a Scraper class
+    try:
+        engine = get_scrapers()[search_engine.lower()]
+    except KeyError:
+        err = [404, 'Incorrect search engine', search_engine]
+        return bad_request(err)
+
+    # get the query parameter
+    query = request.args.get('query')
+    if not query:
+        err = [400, 'Not Found - missing query', query]
+        return bad_request(err)
+
+    # get the output format parameter
+    qformat = request.args.get('format', 'json').lower()
+    if qformat not in ('json', 'xml'):
```

Comment:
We are planning to support CSV format as well, right? 
This will raise an 'Undefined format' exception even for CSV as here we are checking only for JSON and XML format

Reply:
Yes.  Thanks for your review.  I will have to take this PR apart and put it back together again.  There are good ideas here but so much of the code has changed in the past six weeks that a merge is difficult (I have tried a few times).  I will attempt doing so after #424 and #385 land.